ivy/functional/backends/numpy/manipulation.py

```diff
@@ -261,13 +261,13 @@ def unstack(
 
 def clip(
     x: np.ndarray,
-    x_min: Union[Number, np.ndarray],
-    x_max: Union[Number, np.ndarray],
+    x_min: Union[float, int, np.ndarray] = None,
+    x_max: Union[float, int, np.ndarray] = None,
     /,
     *,
     out: Optional[np.ndarray] = None,
 ) -> np.ndarray:
-    return np.asarray(np.clip(x, x_min, x_max, out=out), dtype=x.dtype)
+    return np.clip(x, x_min, x_max, out=out)
```

Comment:
```suggestion
    promoted_type = x.dtype
    if x_min is not None:
        promoted_type = ivy.promote_types(promoted_type, x_min.dtype)
    if x_max is not None:
        promoted_type = ivy.promote_types(promoted_type, x_max.dtype)

    return np.asarray(np.clip(x, x_min, x_max, out=out), dtype= promoted_type)
```

This should fix your failing numpy test.